deeds-tenant-service/src/main/java/io/meeds/tenant/metamask/utils/Utils.java

```diff
@@ -0,0 +1,14 @@
+package io.meeds.tenant.metamask.utils;
+
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+
+public class Utils {
+
+    public static Identity getIdentityByUsername(String name) {
+        IdentityManager identityManager = CommonsUtils.getService(IdentityManager.class);
+        return identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, name);
```

Comment:
```suggestion
        return identityManager.getOrCreateUserIdentity(name);
```

Besides, it's better to use this method call inside the Listener (and add `IdentityManager` as dependency to inject in constructor).